### PR TITLE
feat(env): send a v0/v1 runtime hint on push

### PR DESCRIPTION
**Pull request by kcoopermiller**
## Summary

`prime env push` is v0/v1-blind today. This PR makes it declare which verifiers API the package targets and send that as a top-level `runtime_hint` on both Hub upload calls (`POST /environmentshub/{env_id}/wheels` and `POST /environmentshub/{env_id}/versions`).

- **`--runtime v0|v1`** declares it explicitly.
- Without the flag, the package's own `verifiers` requirement decides (wheel `Requires-Dist`, else pyproject `dependencies`): a lower bound at or above 0.2.0 means v1, any other pin means v0. This is the same fallback the Hub applies server-side to hint-less pushes (platform #4792), so old and new CLIs agree.
- No verifiers requirement, or a URL pin, means nothing is known: the field is **omitted** (not sent as `null`), push warns that the Hub will list the package as Unclassified, and suggests the flag.
- Push prints what it is publishing as, and where that came from, so a mislabel is visible locally.

No source scanning: runtime is declared, not detected. Old servers ignore the unknown field (pydantic default), so this can merge before #4792 deploys; the hint just starts mattering once it does. Needs a prime CLI release (ENG-5767).

---

> [!NOTE]
> **Low Risk**
> Additive CLI metadata on publish with omission when unknown; backward compatible with older Hub APIs, but mis-inferred runtime could mislabel packages once the server consumes the hint.
> 
> **Overview**
> **`prime env push`** now declares which verifiers API (v0 vs v1) a package targets and sends that as top-level **`runtime_hint`** on both Hub upload requests (wheel and source version).
> 
> Users can set it with **`--runtime v0|v1`**. If omitted, the CLI infers runtime from the package’s **`verifiers`** dependency (wheel `Requires-Dist`, else `pyproject` dependencies): a floor **≥ 0.2.0** → v1, any other pin → v0, matching the Hub’s server-side fallback. Missing or URL-only `verifiers` pins omit the field, warn that the Hub will show **Unclassified**, and suggest **`--runtime`**. Push also prints the chosen label and whether it came from the flag or metadata.
> 
> A new **`environment_runtime`** helper encapsulates parsing and classification; tests cover requirement edge cases and invalid **`--runtime`** handling on push.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 05c12a43d7d7c5605cccaf20a3adbb9e09dd31d0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/prime/src/prime_cli/commands/env.py [hunk 1]
```diff
@@ -1082,6 +1095,12 @@ def push(
     """Push environment to registry"""
 
     try:
+        try:
+            declared_runtime = parse_runtime_option(runtime)
+        except ValueError as e:
+            console.print(f"[red]Error: {e}[/red]")
+            raise typer.Exit(1)
```
Comment (mrmoxon) [inline packages/prime/src/prime_cli/commands/env.py]: This `typer.Exit` is caught by the outer `except Exception` at the end of `push`, so `prime env push --runtime v2` prints both the useful validation message and `Unexpected error: 1`. Could we move this validation outside the outer `try`, or explicitly re-raise `typer.Exit` before the generic handler?

**Review by mrmoxon — APPROVED** [review]
Runtime classification matches the Platform contract, both upload payloads carry the same conclusive hint, and inconclusive hints are omitted. Focused tests pass locally. Approved with one small non-blocking CLI error-message comment.

**Review by mrmoxon — DISMISSED** [review]

**Review by mrmoxon — APPROVED** [review]

**Review by kennethnym — APPROVED** [review]
